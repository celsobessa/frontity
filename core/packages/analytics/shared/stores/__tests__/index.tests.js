import Store from '../';
import rawEntities from './raw-entities.json';

describe('Analytics', () => {
  test('action - addCustomDimensions', () => {
    const analytics = Store.create();
    analytics.addCustomDimensions(rawEntities);
    expect(analytics).toMatchSnapshot();
  });

  test('view - customDimensions', () => {
    const analytics = Store.create({
      customDimensionMap: {
        post_33: {
          dimension1: 'dim1',
          dimension2: 'dim2',
        },
      },
    });

    expect(analytics.customDimensions({ type: 'post', id: 34 })).toBe(null);
    expect(
      analytics.customDimensions({ type: 'post', id: 33 }),
    ).toMatchSnapshot();
  });
});
