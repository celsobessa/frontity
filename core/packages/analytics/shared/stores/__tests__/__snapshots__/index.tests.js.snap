// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics action - addCustomDimensions 1`] = `
Object {
  "customDimensionMap": Object {
    "post_33": Object {
      "dimension1": "dim1",
      "dimension2": "dim2",
    },
    "post_36": Object {
      "dimension1": "dim1",
      "dimension2": "dim2",
    },
    "post_42": Object {
      "dimension1": "dim1",
      "dimension2": "dim2",
    },
    "post_54": Object {
      "dimension1": "dim1",
      "dimension2": "dim2",
    },
    "post_57": Object {
      "dimension1": "dim1",
      "dimension2": "dim2",
    },
  },
}
`;

exports[`Analytics view - customDimensions 1`] = `
Object {
  "dimension1": "dim1",
  "dimension2": "dim2",
}
`;
